sawmill-core/src/main/java/io/logz/sawmill/processors/ListIntersectProcessor.java

```diff
@@ -0,0 +1,86 @@
+package io.logz.sawmill.processors;
+
+import io.logz.sawmill.Doc;
+import io.logz.sawmill.ProcessResult;
+import io.logz.sawmill.Processor;
+import io.logz.sawmill.annotations.ProcessorProvider;
+import io.logz.sawmill.utilities.JsonUtils;
+
+import javax.inject.Inject;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+import static org.apache.commons.collections4.CollectionUtils.intersection;
+import static org.apache.commons.collections4.CollectionUtils.isNotEmpty;
+
+@ProcessorProvider(type = "listIntersect", factory = ListIntersectProcessor.Factory.class)
+public class ListIntersectProcessor implements Processor {
+
+    private final String sourceFieldA;
+    private final String sourceFieldB;
+    private final String targetField;
+
+    public ListIntersectProcessor(String sourceFieldA, String sourceFieldB, String targetField) {
+        this.sourceFieldA = sourceFieldA;
+        this.sourceFieldB = sourceFieldB;
+        this.targetField = targetField;
+    }
+
+    @Override
+    public ProcessResult process(Doc doc) {
+        Iterable<Object> listA = doc.getField(sourceFieldA);
+        Iterable<Object> listB = doc.getField(sourceFieldB);
+
+        Set<Object> intersection = new HashSet<>(intersection(listA, listB));
+        if (isNotEmpty(intersection)) {
```

Comment:
Why not create the field and leave the value empty?

Reply:
Seems counter-intuitive to me. But I am not sure what is the common behavior in other processors.

Reply:
Changing it.

Reply:
LGTM.
Note that the automatic build currently does not run all unit tests - this is something @alexpalchuk started working on. So make sure to run all unit tests locally.
Similarly, the automatic upload to the maven repository is also broken...

Reply:
Yes I have already run all tests locally using mvn